rocrate_validator/profiles/five-safes-crate/should/10_outputs.ttl

```diff
@@ -0,0 +1,91 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+@prefix ro: <./> .
+@prefix ro-crate: <https://github.com/crs4/rocrate-validator/profiles/ro-crate/> .
+@prefix five-safes-crate: <https://github.com/eScienceLab/rocrate-validator/profiles/five-safes-crate/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix schema: <http://schema.org/> .
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix validator: <https://github.com/crs4/rocrate-validator/> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+
+
+five-safes-crate:CreateActionHasResultIfActionCompleted
+  a sh:NodeShape ;
+  sh:name "CreateAction" ;
+  sh:description "`CreateAction` with CompleteActionStatus SHOULD have the `schema:result` property." ;
+
+
+  sh:target [
+        a sh:SPARQLTarget ;
+        sh:prefixes ro-crate:sparqlPrefixes ;
+        sh:select """
+            SELECT ?this WHERE {
+              ?this a schema:CreateAction ;
+                      schema:actionStatus "http://schema.org/CompleteActionStatus" .
+            }
+        """
+    ] ;
+
+
+  sh:property [
+    a sh:PropertyShape ;
+    sh:name "Result" ;
+    sh:path schema:result ;
+    sh:minCount 1 ;
+    sh:severity sh:Warning ;
+    sh:message "`CreateAction` with CompleteActionStatus SHOULD have the `schema:result` property." ;
+  ] .
+
+
+
+# This test passes even when it should fail, so it is disabled for now.
+# According to Clause.ai the possible reason for this misbehaviour is that
+# the RO-Crate validator adds inferred types like schema:MediaObject to the
+# entity based on other characteristics (like it being a file). So the entity
+# has BOTH "Filexx" (which doesn't expand to a schema.org URI) AND the inferred
+#  schema:MediaObject, which makes the constraint pass.
+# The fundamental problem is that we can't easily distinguish between
+# user-declared types and validator-inferred types in the materialized RDF graph.
+
+# five-safes-crate:CreateActionResultOutputsHaveAllowedTypes
+#   a sh:NodeShape ;
+#     sh:name "Output" ;
+#     sh:target [
+#       a sh:SPARQLTarget ;
+#       sh:select """
+#         PREFIX schema: <http://schema.org/>
+#         SELECT ?this
+#         WHERE {
+#           ?createAction a schema:CreateAction .
+#           ?createAction schema:result ?this .
+#         }
+#       """ ;
+#     ] ;
+#     sh:message "Every schema:result must point to a defined node (must have an rdf:type)." ;
+#     sh:property [
+#       sh:path rdf:type ;
+#       sh:qualifiedValueShape (
+#         schema:MediaObject
+#         schema:Dataset
+#         schema:Collection
+#         schema:DigitalDocument
+#         schema:PropertyValue
+#       ) ;
+#       sh:qualifiedMinCount 1 ;
```

Comment:
this is a similar mistake to https://github.com/eScienceLab/rocrate-validator/pull/40#discussion_r2577755568 - `qualifiedValueShape` is attempting to check that the values of `rdf:type` (i.e. the entire class objects) follow the shapes listed. So:
1) it's trying to run validation on the class objects themselves rather than checking that the value of `rdf:type` is one of those classes
2) the names that are in the list are not shapes, so have no constraints to apply - that's the reason why this always passes, I think

use `sh:hasValue` and `sh:or` to target the actual value of the path. But needs some care as an entity could have any of these _and_ other types as well.